backend/src/news/router.py

```diff
@@ -142,4 +118,18 @@ def upvote_article(
     :return: JSON 格式的點讚操作狀態訊息
     """
     message = toggle_upvote(article_id, usertoken.id, db)
-    return {"message": message}
\ No newline at end of file
+    return {"message": message}
+@router.post("/news_summary_custom_model")
+async def summarize_news_with_custome_model(payload: NewsSumaryCustomModelSchema, user= Depends(authenticate_user_token)):
+    response = {}
+    if(payload.ai_model=="anthropic") :
+        llm_client=anthropic_client
+    if(payload.ai_model=="openai") :
+        llm_client=openai_client 
+    result = llm_client.generate_summary(payload.content)
```

Comment:
I would have llm_client be custom_llm_client.
You've set llm_client to be openai_client on line 15. If that's the default model, I'm not sure if llm_client = anthropic_client will affect the code above to run differently. Even if it doesn't, better safe than sorry.